Google.Api.Gax.Grpc.GrpcNetClient/GrpcNetClientAdapter.cs

```diff
@@ -0,0 +1,84 @@
+    {
+        // Note: this is "Default" rather than "Instance" as we expect to have other factory methods later, e.g. accepting
+        // an HTTP client factory.
+
+        /// <summary>
+        /// Returns the default instance of this class.
+        /// </summary>
+        public static GrpcNetClientAdapter Default { get; } = new GrpcNetClientAdapter();
+
+        private GrpcNetClientAdapter()
+        {
+        }
+
+        /// <inheritdoc />
+        protected override ChannelBase CreateChannelImpl(string endpoint, ChannelCredentials credentials, GrpcChannelOptions options)
+        {
+            var grpcNetClientOptions = ConvertOptions(credentials, options);
+            var address = ConvertEndpoint(endpoint);
+            return GrpcChannel.ForAddress(address, grpcNetClientOptions);
+        }
+
+        // Internal for testing
+        internal static global::Grpc.Net.Client.GrpcChannelOptions ConvertOptions(ChannelCredentials credentials, GrpcChannelOptions options)
+        {
+            // If service config resolution is explicitly enabled, throw - we can't support that,
+            // and users may be depending on it.
+            if (options.EnableServiceConfigResolution == true)
+            {
+                throw new ArgumentException($"{nameof(options.EnableServiceConfigResolution)} is not currently supported in {nameof(GrpcNetClientAdapter)}");
+            }
+
+            if (options.CustomOptions.Count > 0)
+            {
+                throw new ArgumentException($"{nameof(options.EnableServiceConfigResolution)} are not currently supported in {nameof(GrpcNetClientAdapter)}");
+            }
+
+            // Options we ignore:
+            // - PrimaryUserAgent
+            // - KeepAliveTime
+
+            return new global::Grpc.Net.Client.GrpcChannelOptions
+            {
+                Credentials = credentials,
+                MaxReceiveMessageSize = options.MaxReceiveMessageSize,
+                MaxSendMessageSize = options.MaxSendMessageSize                 
+            };
+        }
+
+        // Regular expression for a simple IPv4 address with optional port. We may want to expand this to handle IPv6 at some point.
+        private static readonly Regex s_simpleAddress = new Regex(@"^\d+\.\d+\.\d+\.\d+(:\d+)?$");
+        // Internal for testing
+        internal static string ConvertEndpoint(string endpoint)
+        {
+            if (endpoint.StartsWith("http:", StringComparison.Ordinal) || endpoint.StartsWith("https:", StringComparison.Ordinal))
+            {
+                return endpoint;
+            }
+            if (s_simpleAddress.IsMatch(endpoint))
+            {
+                // Note that we assume HTTPS for any bare address; this feels like a reasonable assumption for now.
+                return $"https://{endpoint}";
+            }
+            throw new ArgumentException($"Unable to use endpoint '{endpoint}' with Grpc.Net.Client");
```

Comment:
Just a question, if I understand this correctly, something like storage.googleapis.com won't work unless it is prepended with http(s)://?

Reply:
Yikes, you're absolutely right. I need to go back to that, basically.

Reply:
(I've no idea why I thought everything would be numeric.)

Reply:
I'm tempted to just make it add "https://" for anything that doesn't start with "http:" or "https:" and call it done... if that fails (e.g. because we're trying to connect to "ipv6:[::1]:1234") then so be it... Thoughts?

Reply:
That's what I've done for now - PTAL.